### PR TITLE
Fix ambiguity between cast and parenthesized invocation

### PR DESCRIPTION
**Pull request by tamasvajk**
Fixes https://github.com/tree-sitter/tree-sitter-c-sharp/issues/175.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1148,12 +1148,12 @@ module.exports = grammar({
...
     await_expression: $ => prec.right(PREC.UNARY, seq('await', $._expression)),
 
-    cast_expression: $ => prec.right(PREC.CAST, seq(
+    cast_expression: $ => prec.right(PREC.CAST, prec.dynamic(1, seq(
       '(',
       field('type', $._type),
       ')',
       field('value', $._expression)
-    )),
+    ))),
```
[the remark this answers is not in the document]
Comment (damieng) [reply grammar.js:1156]: Well now I'm confused as well - there are a few uses of prec.dynamic and not all of them have conflicts set-up either.

[the remark this answers is not in the document]
Comment (tamasvajk) [reply grammar.js:1156]: I don't really know how to read the output of `tree-sitter parse [FILE] -D`, but `[$._name, $._expression]` conflict seems to be the key for this to happen.

**Review by damieng — APPROVED** [review]

**Comment by damieng** [comment]
I'm okay with merging this even though the relationship behind conflicts vs prec.dynamic aren't clear.

**Comment by tamasvajk** [comment]
> I'm okay with merging this even though the relationship behind conflicts vs prec.dynamic aren't clear.

Sounds good to me.

**Pull request merged**